File: guides/replace-h2-with-real-database-for-testing/postgresqltestresources/java/src/main/java/example/micronaut/Product.java
package example.micronaut;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity // <1>
@Table(name = "products") // <2>
public class Product {

    @Id // <3>
    private Long id;

    @Column(nullable = false, unique = true)  // <4>
    private String code;

    @Column(nullable = false) // <4>
    private String name;

    public Product() {}

    public Product(Long id, String code, String name) {
        this.id = id;
        this.code = code;
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: guides/replace-h2-with-real-database-for-testing/postgresqltestresources/kotlin/src/main/kotlin/example/micronaut/Product.kt
/*
 * Copyright 2017-2026 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Table(name = "products") // <1>
@Entity // <2>
class Product(
    @field:Id // <3>
    var id: Long,

    @field:Column(nullable = false, unique = true) // <4>
    var code: String,

    @field:Column(nullable = false) // <4>
    var name: String
)
